customers/views.py

```diff
@@ -34,24 +36,28 @@ def create():
             "email": f"{email}",
             "dob": f"{dob}",
             "date_created": f"{newCustomer.date_created}",
-        }
+        }, 200
     except exc.IntegrityError:
         # Error Code 409, for resource already exists / duplicate resource / Conflict
         # Error that occurs when a customer already has a place in the database
-        return {"Message": "ERROR, Code: 409, Customer already exists"}
+        # Should occur when a customer enters a unique field such as email and username more than once
+        return {"Message": "ERROR, Code: 409, Customer already exists"}, 409
     except KeyError:
         # Error that occurs when a user leaves fields empty
-        return {"Message": "ERROR, Code: 422, left required fields empty"}
+        return {"Message": "ERROR, Code: 422, left required fields empty"}, 422
 
 
-def retrieve(customer_id=-1):
+@views.route("/list", defaults={"customer_id": None}, methods=["GET"])
+@views.route("/detail/", defaults={"customer_id": None}, methods=["GET"])
+@views.route("/detail/<int:customer_id>", methods=["GET"])
```

Comment:
your routes should not have a path, unless it's the `detail` route - and only then it will be the ID

For example, instead of a `GET` that returns all customers from `/customers/list` it should just be `GET` to `/customers/`

and a `GET` that returns a single customer would be `/customers/1` (if the ID is 1)

Reply:
think about that first comment... with that information, how many route decorators does this function need?